Source/MQTTnet/Client/MqttClient.cs

```diff
@@ -101,7 +101,11 @@ public async Task<MqttClientAuthenticateResult> ConnectAsync(IMqttClientOptions
                 _logger.Info("Connected.");
 
                 Connected?.Invoke(this, new MqttClientConnectedEventArgs(authenticateResult));
-                await ConnectedHandler?.HandleConnectedAsync(new MqttClientConnectedEventArgs(authenticateResult));
+                if(ConnectedHandler != null)
+                {
```

Comment:
Please move the handler to a variable first. Because it may happen that the property is checked in the if statement and then the value is set to null. This is only a theoretical thing but also a common pattern for event handlers (before ? was available).